tests/gene-comparison-tool.spec.ts

```diff
@@ -27,4 +27,15 @@ test.describe('specific viewport block', () => {
     const dropdown = page.locator('#subCategory');
     await expect(dropdown).toHaveText('Targeted Selected Reaction Monitoring (SRM)');
   });
+
+  test('invalid gene results in a 404 redirect', async ({ page }) => {
+    // go to invalid ENSG page
+    await page.goto('/genes/ENSG00000000000');
+
+    // expect a title "to contain" a substring.
+    await expect(page).toHaveTitle('Agora');
+
+    // expect div for page not found content to be visible
+    expect(page.locator('.page-not-found')).toBeVisible();
```

Comment:
I did not write a test to check the url as the current behavior is not to change the url when redirecting.  For example navigating to: /genes/ENSG00000 does not change the URL to: /404-not-found.  Therefore, div content on the 404 page is checked to ensure that the 404 page is loaded.